src/utils/parse-user-agent.ts

```diff
@@ -0,0 +1,51 @@
+import Bowser from 'bowser';
+
+const DEFAULT_BROWSER_NAME    = 'Other';
+const DEFAULT_BROWSER_VERSION = '0.0';
+// NOTE: we need to pass the ' ' user agent to get a "default" parsed object (GH-481)
+const EMPTY_BOWSER_USER_AGENT = ' ';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export default function parseUserAgent (userAgent: string, browserAlias?: string): any {
+    let parsedUserAgent;
```

Comment:
The `browserAlias` parameter is necessary only to return the `.headless` property from the `parseUserAgent` function.
I propose to split this function into 2 two parts:
* `parseUserAgent(userAgent: string)` - parse user agent string and returns the parsed part.
* `isHeadless(alias: string)` - calculated that browser is running in the headless mode or not.

and call both these function in `BrowserConnection.establish` method.

@AndreyBelym What do think about it?

Reply:
Agreed. Ideally we should use `provider.isHeadlessBrowser()` method to determine if the current browser is headless:

https://github.com/DevExpress/testcafe/blob/master/src/browser/provider/index.ts#L235

`BrowserConnection` can retrieve it immediately after a browser has opened, and `t.browser` can retrieve it via `testRun.browserConnection.headless`.